src/wp-includes/rest-api/endpoints/class-wp-rest-template-revisions-controller.php

```diff
@@ -0,0 +1,298 @@
+					'permission_callback' => array( $this, 'delete_item_permissions_check' ),
+					'args'                => array(
+						'force' => array(
+							'type'        => 'boolean',
+							'default'     => false,
+							'description' => __( 'Required to be true, as revisions do not support trashing.' ),
+						),
+					),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Get the parent post, if the ID is valid.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param int $parent_post_id Supplied ID.
+	 * @return WP_Post|WP_Error Post object if ID is valid, WP_Error otherwise.
+	 */
+	protected function get_parent( $parent_post_id ) {
+		$template = get_block_template( $parent_post_id, $this->parent_post_type );
+
+		if ( ! $template ) {
+			return new WP_Error(
+				'rest_post_invalid_parent',
+				__( 'Invalid template parent ID.' ),
+				array( 'status' => 404 )
+			);
+		}
+
+		return get_post( $template->wp_id );
+	}
+
+	/**
+	 * Prepares the item for the REST response.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param mixed           $item    WordPress representation of the item.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response Response object.
+	 */
+	public function prepare_item_for_response( $item, $request ) {
+		$template = _build_block_template_result_from_post( $item );
+		$response = $this->parent_controller->prepare_item_for_response( $template, $request );
+
+		$fields = $this->get_fields_for_response( $request );
+		$data   = $response->get_data();
+
+		if ( in_array( 'parent', $fields, true ) ) {
+			$data['parent'] = (int) $item->post_parent;
+		}
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		// Wrap the data in a response object.
+		$response = new WP_REST_Response( $data );
+
+		if ( rest_is_field_included( '_links', $fields ) || rest_is_field_included( '_embedded', $fields ) ) {
+			$links = $this->prepare_links( $template );
+			$response->add_links( $links );
+		}
+
+		return $response;
+	}
+
+	/**
+	 * Checks if a given request has access to delete a revision.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has access to delete the item, WP_Error object otherwise.
+	 */
+	public function delete_item_permissions_check( $request ) {
+		$parent = $this->get_parent( $request['parent'] );
+		if ( is_wp_error( $parent ) ) {
+			return $parent;
+		}
+
+		if ( ! current_user_can( 'delete_post', $parent->ID ) ) {
+			return new WP_Error(
+				'rest_cannot_delete',
+				__( 'Sorry, you are not allowed to delete revisions of this post.' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		$revision = $this->get_revision( $request['id'] );
+		if ( is_wp_error( $revision ) ) {
+			return $revision;
+		}
+
+		if ( ! current_user_can( 'edit_theme_options' ) ) {
```

Comment:
should they need delete_themes to delete? or is this copied from the main controller?

Reply:
Copied from main controller.